[OSDOCS-4235]: CPMS resiliency and recovery (#53246)
openshift/openshift-docs: modules/cpmso-failure-domains-balancing.adoc

// Module included in the following assemblies:
//
// * machine_management/cpmso-resiliency.adoc

:_content-type: CONCEPT
[id="cpmso-failure-domains-balancing_{context}"]
= Balancing control plane machines

The control plane machine set balances control plane machines across the failure domains that are specified in the custom resource (CR).

//If failure domains must be reused, they are selected alphabetically by name. 
When possible, the control plane machine set uses each failure domain equally to ensure appropriate fault tolerance. If there are fewer failure domains than control plane machines, failure domains are selected for reuse alphabetically by name. For clusters with no failure domains specified, all control plane machines are placed within a single failure domain. 

> Some changes to the failure domain configuration cause the control plane machine set to rebalance the control plane machines. For example, if you add failure domains to a cluster with fewer failure domains than control plane machines, the control plane machine set rebalances the machines across all available failure domains. 

Reviewer: ```suggestion
Changes to the failure domain configuration can cause the control plane machine set to rebalance the control plane machines. For example, adding failure domains to a cluster with fewer failure domains than control plane machines can cause the control plane machine set to rebalance machines across all available failure domains.
```
